src/include/daos_fs_sys.h

```diff
@@ -0,0 +1,454 @@
+/**
+ * (C) Copyright 2018-2021 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+/**
+ * \file
+ *
+ * DAOS File System "Sys" API
+ *
+ * The DFS Sys API provides a simplified layer directly on top
+ * of the DFS API that is more similar to the equivalent
+ * POSIX libraries.
+ */
+
+#include <dirent.h>
+
+#include "daos.h"
+#include "daos_fs.h"
+
+#define DFS_SYS_NO_CACHE 1
+#define DFS_SYS_NO_LOCK 2
+
+/* TODO - Not necessary, but it would be useful if we keep "dfs"
+ * public in this struct in case a user wants to call a dfs function
+ * directly. The main use case for this is that the dfs functions
+ * operate directly on an object instead of a path, so if the user
+ * already has the object handle, it's technically more efficient
+ * to call the dfs functions directly.
+ * This is mostly a way of future-proofing, in case some new functions are
+ * added to dfs.
+ * An alternative is that we could just add direct wrappers for each of
+ * the dfs functions, but we would get some name conflicts, since dfs_sys
+ * already adds wrappers for most of these functions that take a "path".
+ * As an example, see dfs_sys_punch and dfs_sys_opunch below.
+ */
+/** struct holding attributes for the dfs_sys calls */
+typedef struct dfs_sys {
```

Comment:
Not sure about this one. Maybe it just need approval?

Reply:
This one is incompatible with our coding style, you could simply drop the typedef but then users would need to use "strut dfs_sys" rather than dfs_sys_t.  I think just living with it is the best option.

Reply:
I'm not sure I follow. How do we have other typedefs if it's against our coding style?

Reply:
you can ignore this